Framework/DataObjects/src/TimeSplitter.cpp

```diff
@@ -678,5 +678,58 @@ void TimeSplitter::splitEventVec(const std::function<const DateAndTime(const Eve
   }
 }
 
+/**
+ * @brief Calculate the target indices for a given set of times.
+  Returns a vector of pairs, where each pair contains a target index and a pair of size_t representing the start and
+ stop indices in the input times vector that correspond to that target.
+
+  If the TimeSplitter is empty, the entire range of times is returned with NO_TARGET.
+ *
+ * @param times
+ * @return std::vector<std::pair<int, std::pair<size_t, size_t>>>
+ */
+std::vector<std::pair<int, std::pair<size_t, size_t>>>
+TimeSplitter::calculate_target_indices(const std::vector<DateAndTime> &times) const {
+  std::vector<std::pair<int, std::pair<size_t, size_t>>> indices;
+  if (this->empty()) {
+    // if the TimeSplitter is empty, then we just return the whole range with NO_TARGET
+    indices.emplace_back(NO_TARGET, std::make_pair(0, times.size()));
+    return indices;
+  }
+  indices.reserve(m_roi_map.size());
```

Comment:
My experience is that if you have the constructor create all the values and change them later, it is significantly faster than `reserve` + `emplace_back`. It probably has to do with allocating little objects all over the place.

Reply:
Taking a look at this again I realized I am reversing too much because of skipping the NO_TARGET. Then I realized this could all be simplified greatly by using `getSplittingIntervals`.